recipes/repo.rb

```diff
@@ -142,6 +142,34 @@
   action [:manage]
 end
 
+# Remove previous GPG deployment
+execute "gpg-agent --daemon" do
+  user agent_username
+  group agent_username
+  environment 'HOME' => "/home/#{agent_username}"
+  only_if { ::File.exist?("/home/#{agent_username}/.ssh/gpg_private_key.sec") }
+  not_if "gpg-agent"
+end
+execute "(echo y) | gpg --command-fd 0 --no-tty --yes --delete-secret-keys #{gpg_key['fingerprint']}" do
+  user agent_username
+  group agent_username
+  environment 'HOME' => "/home/#{agent_username}"
+  only_if { ::File.exist?("/home/#{agent_username}/.ssh/gpg_private_key.sec") }
+  only_if "gpg --list-secret-keys #{gpg_key['fingerprint']}"
```

Comment:
Can multiple `only_if`s be used here and both work? I've not seen any documentation that says they stack or don't.

Reply:
I'm actually not sure. It did what I wanted it to do, so I just assumed.

How can this be written as a compound conditional?

Reply:
```suggestion
  only_if "test -f /home/#{agent_username}/.ssh/gpg_private_key.sec && gpg --list-secret-keys #{gpg_key['fingerprint']}"
```

Reply:
> I think the most straightforward way is to test file existence in shell rather than with the Ruby methods.

Ah, good call.